### PR TITLE
build: Fix rpserver and rpproxy CMake builds

### PR DESCRIPTION
**Pull request by Jasinsky**
## Summary

This fixes missing include directories for iperf application.

Make based build was able to proceed - CMake build had two aforementioned issues.

## Impact

Hard to estimate.

## Testing

Local playing around rpserver and rpproxy in sim flavor.

### CODE REVIEW

**Comment by Jasinsky** [comment]
For your visibility @acassis @xiaoxiang781216

File: netutils/iperf/CMakeLists.txt [hunk 1]
```diff
@@ -30,7 +30,9 @@ if(CONFIG_NETUTILS_IPERF)
     STACKSIZE
     ${CONFIG_NETUTILS_IPERF_STACKSIZE}
     PRIORITY
-    ${CONFIG_NETUTILS_IPERF_PRIORITY})
+    ${CONFIG_NETUTILS_IPERF_PRIORITY}
+    INCLUDE_DIRECTORIES
+    ${CMAKE_BINARY_DIR}/apps/include)
```
Comment (xiaoxiang781216) [inline netutils/iperf/CMakeLists.txt:35]: should  `${CMAKE_BINARY_DIR}/apps/include` add to the search path by default? @xuxin930

File: system/argtable3/CMakeLists.txt [hunk 2]
```diff
@@ -82,6 +82,11 @@ if(CONFIG_SYSTEM_ARGTABLE3)
   target_sources(argtable3 PRIVATE ${SRCS})
 
   if(NOT EXISTS ${CMAKE_BINARY_DIR}/apps/include/argtable3.h)
+
+    if(NOT EXISTS ${CMAKE_BINARY_DIR}/apps/include)
```
Comment (xiaoxiang781216) [inline system/argtable3/CMakeLists.txt:86]: should we create `${CMAKE_BINARY_DIR}/apps/include` by default? @xuxin930

File: netutils/iperf/CMakeLists.txt (hunk 1 above)
Comment (Jasinsky) [reply netutils/iperf/CMakeLists.txt:35]: Good question, wanted to have minimal changes here.
Will wait for @xuxin930 response - or if I will discover any other default configuration relying on build folder include directory I will make it invariant.

File: system/argtable3/CMakeLists.txt (hunk 2 above)
Comment (Jasinsky) [reply system/argtable3/CMakeLists.txt:86]: Good question, wanted to have minimal changes here.
Will wait for @xuxin930 response - or if I will discover any other default configuration relying on build folder include directory I will make it invariant.

**Comment by xuxin930** [comment]
hi @Jasinsky @xiaoxiang781216 nice catch!
argtable3/src is set by Make.defs and takes effect in the whole context of apps.
lets make it takes effect globally in CMake by.
```cmake
# in argtable3 CMakeLists
  set_property(
    TARGET nuttx
    APPEND
    PROPERTY NUTTX_INCLUDE_DIRECTORIES 
    ${NUTTX_APPS_DIR}/system/argtable3/argtable3/src)
```

File: system/argtable3/CMakeLists.txt (hunk 2 above)
Comment (Jasinsky) [reply system/argtable3/CMakeLists.txt]: Done as xuxin930 suggested.

File: netutils/iperf/CMakeLists.txt (hunk 1 above)
Comment (Jasinsky) [reply netutils/iperf/CMakeLists.txt]: Done as xuxin930 suggested.

**Comment by Jasinsky** [comment]
@xiaoxiang781216  @xuxin930 - Thank you for your suggestion, applied - df744eb

**Review by acassis — APPROVED** [review]

**Pull request merged**